Documentation/observability/visibility.rst

```diff
@@ -12,63 +12,74 @@ Layer 7 Protocol Visibility
 
 .. note::
 
-    This feature requires enabling L7 Proxy support. Without it, the visibility annotation is ignored.
+    This feature requires enabling L7 Proxy support.
 
-While :ref:`monitor` provides introspection into datapath state, by default it
-will only provide visibility into L3/L4 packet events. If :ref:`l7_policy` are
-configured, one can get visibility into L7 protocols, but this requires the full
-policy for each selected endpoint to be written. To get more visibility into the
-application without configuring a full policy, Cilium provides a means of
-prescribing visibility via `annotations <https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/>`_
-when running in tandem with Kubernetes.
+While :ref:`monitor` provides introspection into datapath state, by default, it
+will only provide visibility into L3/L4 packet events. If you want L7
+protocol visibility, you can use L7 Cilium Network Policies (see :ref:`l7_policy`).
 
-Visibility information is represented by a comma-separated list of tuples in
-the annotation:
 
-``<{Traffic Direction}/{L4 Port}/{L4 Protocol}/{L7 Protocol}>``
-
-For example:
-
-::
-
-  <Egress/53/UDP/DNS>,<Egress/80/TCP/HTTP>
-
-
-To do this, you can provide the annotation in your Kubernetes YAMLs, or via the
-command line, e.g.:
-
-.. code-block:: shell-session
-
-    kubectl annotate pod foo -n bar policy.cilium.io/proxy-visibility="<Egress/53/UDP/DNS>,<Egress/80/TCP/HTTP>"
+.. note::
 
-Cilium will pick up that pods have received these annotations, and will
-transparently redirect traffic to the proxy such that the output of
-``cilium monitor`` shows traffic being redirected to the proxy, e.g.:
+    Historically, it had been possible to enable L7 visibility using Pod
+    annotations (``policy.cilium.io/proxy-visibility``). This method is
+    no longer supported and we recommend users to switch to L7 policies instead.
+
+The following example demonstrates how to enable visibility for all DNS (UDP/53)
+and HTTP (ports TCP/80 and TCP/8080) traffic within the ``default`` namespace. 
+You do that by creating a ``CilimNetworkPolicy`` resource containing two L7 rules 
+-- one for DNS and one for HTTP traffic. If all you're interested in is visibility, 
+you can omit or wildcard the L7 matching conditions as it is shown below:
```

Comment:
I have a suggestion for rewording this a little to pad it out. I think it also makes sense to make it more clear that when we omit or wildcard these rules, we are explicitly permitting all traffic that matches the ports in those rules.

```suggestion
To enable visibility for L7 traffic, create a ``CiliumNetworkPolicy`` that specifies
L7 rules. Traffic matching a L7 rule in a ``CiliumNetworkPolicy`` will become
visible. The following example enables visibility for DNS (UDP/53) and HTTP
(ports TCP/80 and TCP/8080) traffic within the ``default`` namespace by
specifying two L7 rules -- one for DNS and one for HTTP. L7 matching
conditions on the rules have been omitted or wildcarded, which will
permit all requests that match the L4 section of each rule:
```